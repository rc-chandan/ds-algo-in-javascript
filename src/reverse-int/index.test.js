// @ts-check

const reverseInt = require('./');

describe('Reverse Integer', () => {
  it('reverseInt is a function', () => {
    expect(typeof reverseInt).toBe('function');
  });
  it('reverse positive integer', () => {
    expect(reverseInt(123456)).toBe(654321);
  });

  it('reverse negative integer', () => {
    expect(reverseInt(-6738)).toBe(-8376);
  });

  it('throw error for non integer numbers', () => {
    expect(() =>
      reverseInt(Number.MAX_SAFE_INTEGER + 1)
    ).toThrowErrorMatchingSnapshot();
  });
});
