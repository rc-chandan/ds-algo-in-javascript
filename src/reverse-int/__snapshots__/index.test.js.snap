// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reverse Integer throw error for non integer numbers 1`] = `"Not an integer"`;
